sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -771,7 +777,11 @@ object SparkSession {
 
           val sparkConf = new SparkConf()
           options.foreach { case (k, v) => sparkConf.set(k, v) }
-          SparkContext.getOrCreate(sparkConf)
+          val sc = SparkContext.getOrCreate(sparkConf)
+          // maybe this is an existing SparkContext, update its SparkConf which maybe used
```

Comment:
We can check whether the sc was pre-existing, right ?

In that case the foreach below is not needed.